src/librustc_errors/diagnostic.rs

```diff
@@ -120,6 +120,9 @@ impl Diagnostic {
     }
 
     /// Adds a span/label to be included in the resulting snippet.
+    /// This label will be shown together with the original span/label used when creating the
+    /// diagnostic, *not* a span added by one of the `span_*` methods.
+    ///
     /// This is pushed onto the `MultiSpan` that was created when the
     /// diagnostic was first built. If you don't call this function at
     /// all, and you just supplied a `Span` to create the diagnostic,
```

Comment:
I have no idea what this second paragraph here means. I left it because maybe it helps someone else? But it seems to only be helpful for people that actually know the internals of how diagnostics work. *Maybe* it even says what I am now also saying in the new sentence I added? I couldn't tell.^^

Reply:
If you called `struct_span_err`, for example, you end up with an error with an underline under that span with no label. If you use `span_label` on the `DiagnosticBuilder` you can either set the label of the primary span or add more spans to the diagnostic. I believe this is what the comment is trying to convey.

Reply:
Okay, that wasn't really clear to me as I never heard of `MultiSpan`. Now it says the same thing in two different ways then, maybe that helps...

Reply:
`MultiSpan` is the only way of setting multiple primary spans, conceptually just `Vec<Span>`.

Reply:
I see. I thought it would refer to how `span_note` also adds a new span to the message, but that seems to be a different kind of "multiple spans".